datadog/resource_datadog_service_level_objective.go

```diff
@@ -0,0 +1,433 @@
+package datadog
+
+import (
+	"fmt"
+	"log"
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/zorkian/go-datadog-api"
+)
+
+func resourceDatadogServiceLevelObjective() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDatadogServiceLevelObjectiveCreate,
+		Read:   resourceDatadogServiceLevelObjectiveRead,
+		Update: resourceDatadogServiceLevelObjectiveUpdate,
+		Delete: resourceDatadogServiceLevelObjectiveDelete,
+		Exists: resourceDatadogServiceLevelObjectiveExists,
+		Importer: &schema.ResourceImporter{
+			State: resourceDatadogServiceLevelObjectiveImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			// Common
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				StateFunc: func(val interface{}) string {
+					return strings.TrimSpace(val.(string))
```

Comment:
Any specific reason for this? Not against it but curious 😄 
I guess a typo of "A gooddescription" -> "A good description" wouldn't get picked up in this case if I understand correctly?

Reply:
https://golang.org/pkg/strings/#TrimSpace only trims leading and trailing space